pandas/core/window/rolling.py

```diff
@@ -1409,6 +1409,16 @@ def quantile(self, quantile: float, interpolation: str = "linear", **kwargs):
 
         return self._apply(window_func, name="quantile", **kwargs)
 
+    def rank(self, pct: bool = False, method: str = "average", ascending: bool = True, **kwargs):
```

Comment:
Nit: Could you have this order mirror `DataFrame.rank` as close as possible i.e. `rank(self, method, ascending, pct)`

Also is `na_option` applicable here like in `DataFrame.rank`: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.rank.html

Reply:
I see I think `na_options` may be hard to resolve with `min_periods` since both key words are supposed to handle NaNs, so maybe it's okay to leave out `na_options` for now.

The `rolling.apply` behavior is expected due to the `min_periods` behavior

```
min_periods : int, default None

    Minimum number of observations in window required to have a value (otherwise result is NA). For a window that is specified by an offset, min_periods will default to 1. Otherwise, min_periods will default to the size of the window.

```